site/source/tutorials/1020_registries/index.md

```diff
@@ -0,0 +1,130 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:43-86 %}
+
+Notice that we are passing a generic type to the `w()` function call, this is because when using a `registry` label it is unable to infer the properties interface. As a result falls back to the default, `WidgetProperties` interface. Passing the generic tells `w()` the type of widget the label represents in the `registry` and will correctly enforce the widget's properties interface.
+
+Next, we'll create a widget that is lazily loaded when needed!
+
+{% section %}
+
+## Lazy Loading Widgets
+
+First we need to extract the `_renderBack` function from `Worker.ts` into a new widget `WorkerBack.ts` and then add the new widget to the registry.
+
+{% task 'Add the following code in `WorkerBack.ts`' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerBack.ts' %}
+
+{% task 'Update the `_renderBack` function to use the `WorkerBack` registry items in `Worker.ts`' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Worker.ts' lines:51-65 %}
+
+{% aside 'Use Before You Define' %}
+The `registry` is designed to mirror the behavior and API of custom elements where ever possible. One neat feature is that a registry item can used before it is defined and, once defined, widgets that use the it will automatically re-render!
+{% endaside %}
+
+Now we need to add the `registry` definition for `WorkerBack.ts` to lazily load when it is required. To do this instead of adding a concrete widget class, we add a function that when called, requires and returns the concrete widget. This function will not be called until the first time the application tries to use the widget label as part of the usual `render` cycle. Initially before the widget has loaded, nothing will be rendered. Once it has loaded any widgets that use the lazy widget will automatically re-render.
+
+{% task 'Import `load` from `@dojo/core/load` in `main.ts`' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:1 %}
+
+{% task 'Define a `const` for `require` that is available at runtime' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:13 %}
+
+{% task 'Now define a function that lazily loads `WorkerBack.ts` in the registry' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:24-26 %}
+
+Now that the `WorkerBack` widget is defined to lazily load, running the application with the browser developer tools open should show that the `WorkerBack.js` file is only loaded when a worker card is click on for the first time:
+
+{% aside 'Auto Bundling Support' %}
+To fully support lazy loading, the Dojo CLI build command will automatically bundle any lazily defined widgets into their own separate file! To learn more, take a look at the [build command tutorial](../comingsoon.html).
+{% endaside %}
+
+![lazy loading gif](./images/lazy-load-widget.gif)
+
+{% section %}
+
+## Summary
+
+Put a summary here.
```

Comment:
Needs a summary :)

Reply:
hehe I hadn't got round to that quite yet.